includes/wikia/services/WikiService.class.php

```diff
@@ -262,43 +262,46 @@ public function getSiteStats( $wikiId = 0 ) {
 	}
 
 	/**
-	 * get list of top editors
+	 * get list of top editors (bots excluded)
 	 *
 	 * @param integer $wikiId
 	 * @param integer $limit
-	 * @param bool    $excludeBots
 	 *
 	 * @return array topEditors [ array( user_id => edits ) ]
 	 */
-	public function getTopEditors( $wikiId = 0, $limit = 30, $excludeBots = false ) {
+	public function getTopEditors( int $wikiId , int $limit ) {
 		wfProfileIn( __METHOD__ );
 
-		$wikiId = ( empty($wikiId) ) ? $this->wg->CityId : $wikiId ;
 		$fname = __METHOD__;
 
 		$topEditors = WikiaDataAccess::cache(
-			wfSharedMemcKey( 'wiki_top_editors', $wikiId, $excludeBots ),
+			wfSharedMemcKey( 'wiki_top_editors:v1', $wikiId ),
 			static::TOPUSER_CACHE_VALID,
-			function() use ( $wikiId, $excludeBots, $fname ) {
-				global $wgSpecialsDB;
+			function() use ( $wikiId, $fname ) {
+				global $wgDWStatsDB;
 				$topEditors = array();
 
-				$db = wfGetDB( DB_SLAVE, array(), $wgSpecialsDB );
+				$db = wfGetDB( DB_SLAVE, array(), $wgDWStatsDB );
 
 				$result = $db->select(
-					array( 'events_local_users' ),
-					array( 'user_id', 'edits', 'all_groups' ),
-					array( 'wiki_id' => $wikiId, 'edits != 0' ),
+					array( 'rollup_wiki_user_events' ),
+					array( 'user_id', 'SUM(edits + creates) as edits' ),
+					array(
+						'period_id' => DataMartService::PERIOD_ID_WEEKLY,
```

Comment:
Isn't this where we wanted to use monthly period?

Reply:
But it's not yet there. Weekly period works quite nice though (from performance point of view). Let's use it before `period_id = 3` data is there for the last year.